### DISCUSSION TITLE
Convert 600x480+24 GIF to 1280x1024+32

### DISCUSSION THREAD

**Discussion by paulmenzel** [ImageMagick/ImageMagick#5501] [Digital Image Processing]
[There is a SeaBIOS patch, adding GIF bootsplash support.](https://mail.coreboot.org/hyperkitty/list/seabios@seabios.org/thread/NNAQ46SOWCBKFVY34CHVKQBI5RJZN7AY/) The [example GIF bootsplash][https://michael.bacarella.com/bootsplash.gif) is:

    $ identify bootsplash.gif
    bootsplash.gif[0] GIF 640x480 640x480+0+0 8-bit sRGB 256c 0.000u 0:00.000
    bootsplash.gif[1] GIF 640x480 640x480+0+0 8-bit sRGB 256c 0.000u 0:00.000
    bootsplash.gif[2] GIF 640x480 640x480+0+0 8-bit sRGB 256c 640039B 0.000u 0:00.000

First question: Where can I see the image depth in the output? It should be 24 bits for the file.

As there is no corresponding video mode supported by my hardware, I am trying to get a GIF file for the modes below like:

    0x110  640 x  480 x 16 (1280)  Direct color, mask: 5/5/5/0  pos: 10/5/0/0
    0x111  640 x  480 x 16 (1280)  Direct color, mask: 5/6/5/0  pos: 11/5/0/0

    0x124 1280 x 1024 x 32 (5120)  Direct color, mask: 8/8/8/0  pos: 16/8/0/0

But converting the GIF seems to fail:

    $ convert bootsplash.gif -resize 1280x1024+32 bootsplash2.gif
    $ identify bootsplash2.gif
    bootsplash2.gif[0] GIF 1280x960 1280x960+0+0 8-bit sRGB 256c 0.000u 0:00.000
    bootsplash2.gif[1] GIF 1280x960 1280x960+0+0 8-bit sRGB 256c 0.000u 0:00.000
    bootsplash2.gif[2] GIF 1280x960 1280x960+0+0 8-bit sRGB 256c 2.05915MiB 0.000u 0:00.000

Using `-depth 32` also didn’t work.

How can I achieve the wanted results? Thank you in advance.

**Comment by snibgo** [comment]
> First question: Where can I see the image depth in the output? It should be 24 bits for the file.

I suppose your "24 bits for the file" means three channels, each of 8 bits. GIF files have palettes of up to 256 entries. Each palette entry has three channels (for R, G and B), each of 8 bits. One entry may be transparent instead of an opaque colour. See https://en.wikipedia.org/wiki/GIF for more details.

> I am trying to get a GIF file for the modes below like:
0x110  640 x  480 x 16 (1280)  Direct color, mask: 5/5/5/0  pos: 10/5/0/0
0x111  640 x  480 x 16 (1280)  Direct color, mask: 5/6/5/0  pos: 11/5/0/0
0x124 1280 x 1024 x 32 (5120)  Direct color, mask: 8/8/8/0  pos: 16/8/0/0

I don't know what those numbers mean.

You can resize an image with `-resize WxH` where W and H are numbers for width and height, for example `-resize 1280x1024`. See https://imagemagick.org/script/command-line-options.php#resize . Other operations to resize an images include `-sample`, `-scale` and `-distort resize`.

`-depth 32` means you want 32 bits per channel per pixel. But GIF can't record that many bits per channel per pixel. You could use, for example, `-depth 3` to *reduce* the number of bits per channel per pixel to 3. But for a GIF file, we can't record more than 8.
